lib/new_relic/agent/datastores/nosql_obfuscator.rb

```diff
@@ -0,0 +1,41 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    module Datastores
+      module NosqlObfuscator
+        ALLOWLIST = [:operation].freeze
+
+        def self.obfuscate_statement(source, allowlist = ALLOWLIST)
+          if source.is_a?(Hash)
+            obfuscated = {}
+            source.each do |key, value|
+              if allowlist.include?(key)
+                obfuscated[key] = value
+              else
+                obfuscated[key] = obfuscate_value(value, allowlist)
+              end
+            end
+            obfuscated
+          else
+            obfuscate_value(source, allowlist)
+          end
+        end
+
+        QUESTION_MARK = '?'.freeze
```

Comment:
I realize this isn't a new line, but just showed up as such from the rename - but this `#freeze` operation is redundant. The magic comment at the top is already freezing the string.

Reply:
I bet that's originally from before that was a thing. I believe 2.2 doesn't recognize the frozen_string_literal comment though, so we might want to keep it around a little while longer till we drop support for 2.2?

Reply:
Fair enough. I think anyone tolerating Ruby 2.2 performance in 2022 won't gain much from having a string frozen 😄 , but I'm ok with keeping it around up until there's a linter warning about the redundancy (I think there is one that exists, but we must have it disabled for now).